app/views/spree/admin/users/_api_fields.html.haml

```diff
@@ -1,10 +1,15 @@
 %fieldset.omega.six.columns
   %legend= t('spree.api.access')
+  = form_tag spree.admin_user_path(@user), method: :put do
+    = check_box_tag :show_api_key_view, 1, @user.show_api_key_view, onchange: "this.form.submit()"
+    = label_tag :show_api_key_view, t('spree.api.toggle_api_key_view')
+    %input(type="hidden" name="user[show_api_key_view]" value="#{!@user.show_api_key_view}")
```

Comment:
It might be worth trying Rails' default form helpers which would add the hidden field and the value automatically. I know that the rest of the page doesn't use it but it's very old code. Would you mind trying that out?

Reply:
Yeah for sure. I extracted all buttons in separated rails forms. They call now the update action.